### ISSUE TITLE
Version 4.10.1 trying to access "jobs" under extensions - why?

### ISSUE THREAD

**Issue by rwmajor2** [fabric8io/kubernetes-client#2300]
I am trying to use this client with a deployment in OpenShift.  As we know, extensions::jobs has been deprecated.  However, when we try to apply our YAML, I keep getting this error message:

Admin|1261|1|20|||com.arcgis.admin.site.SiteManager|Relational Datastore Module Installation Failed. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://10.96.0.1/apis/batch/v1/namespaces/xyz/jobs/relds-aqz9m. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. jobs.extensions "relds-aqz9m" is forbidden: User "system:serviceaccount:arcgis:enterpriseadmin-serviceaccount" cannot get resource "jobs" in API group "extensions" in the namespace "xyz".

The Job YAML is using batch/v1.  Any thoughts?  Extensions::jobs does indeed _not_ exists in Openshift, but I am curious why this client is trying to check that end point.

**Comment by rohanKanojia** [comment]
Are you sure your `ServiceAccount` has correct permissions?

**Comment by rwmajor2** [comment]
This is what we have.

```
rules:
  - apiGroups: [""]
    resources: ["secrets", "configmaps", "services"]
    verbs: ["get", "create", "update", "delete"]
  - apiGroups: [""]
    resources: ["persistentvolumeclaims"]
    verbs: ["create", "delete"]
  - apiGroups: ["apps"]
    resources: ["statefulsets", "deployments", "replicasets"]
    verbs: ["get", "list", "create", "update", "patch", "delete"]
  - apiGroups: ["apps"]
    resources: ["deployments/status"]
    verbs: ["get", "list"]
  - apiGroups: ["extensions"]
    resources: ["ingresses", "deployments"]
    verbs: ["get", "create", "update", "delete"]
  - apiGroups: ["extensions"]
    resources: ["deployments/status"]
    verbs: ["get","list"]
  - apiGroups: ["batch"]
    resources: ["jobs"]
    verbs: ["get", "create", "update", "delete"]
  - apiGroups: ["autoscaling"]
    resources: ["horizontalpodautoscalers"]
    verbs: ["create", "update", "get", "delete"]
```

**Comment by rohanKanojia** [comment]
looks okay to me. Our `OpenShiftClient` is based on `KubernetesClient`, I suspect that while loading yaml, handlers are present for both OpenShift and Kubernetes resources. Maybe that's why it's hitting Job endpoint.

**Comment by rwmajor2** [comment]
@rohanKanojia Thanks for the feedback.  Actually, it fails for OpenShift and a opensource K8s deployment also.

**Comment by manusa** [comment]
Which OpenShift version are you testing this with?

**Comment by itsmurugappan** [comment]
we have the same issue, with kubernetes 1.16.8 where jobs belong group "batch/v1". Not sure why I need to give access to "extensions" group.

**Comment by rohanKanojia** [comment]
@itsmurugappan: Hi, could you please share a reproducer with us so that we can investigate?

**Comment by rhuss** [comment]
For me it looks like the client assumes a very old OpenShift cluster, where the `Job` resources where indeed maintained in the `extensions` group. Actually `batch/v1` has been introduced with Kubernetes 1.2 and the `extensions/v1beta1` Job has been removed in Kubernetes 1.6. See https://github.com/kubernetes/kubernetes/issues/32763 for details.

OpenShift 3.5 was the last version of OpensShift that supported the job in the extensions group. Since then only `batch/v1` jobs are available.

I assume its safe to switch all usages for Jobs in `extensions/v1beta1` to `batch/v1` (if the support map allows this).

**Comment by rhuss** [comment]
Jobs in `batch/vi` are available since OpenShift 3.2, so if there is no support for OpenShift 3.0 or 3.1 in the kubernetes-client, then its IMO save to switch unconditionally to `batch/v1` for the Jobs resource.

**Comment by rohanKanojia** [comment]
Thanks a lot for looking into this. I think this is being caused by our `BackwardCompatibilityInterceptor` which converts failed 404s for `batch/v1` to `extension/v1beta1`. I think removing this should fix client to use `batch/v1` Job always:

https://github.com/fabric8io/kubernetes-client/blob/ec1e20eca8996db6a6ed0b208c8e04da1b7b7b8b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/BackwardsCompatibilityInterceptor.java#L105-L106

**Comment by rohanKanojia** [comment]
Would anyone like to create a PR for this? Just need to remove this line from Interceptor and verify

**Comment by itsmurugappan** [comment]
i can do it.

**Comment by rohanKanojia** [comment]
@itsmurugappan : awesome, Thanks a lot for taking time to fix it!

**Comment by itsmurugappan** [comment]
https://github.com/fabric8io/kubernetes-client/pull/2585

**Comment by rohanKanojia** [comment]
Closed via #2585
